tapestry-plugin-updates.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+function accordion_update() {
+    $query_args = array(
+        'post_type' => 'tapestry_node',
+        'meta_query' => array(
+            array(
+               'key'     => 'tapestry_node_data',
+               'value'   => '"mediaType";s:9:"accordion"',
+               'compare' => 'LIKE',
+            ),
+        ),
+    );
+    $query = new WP_Query( $query_args );
+    while ( $query->have_posts() ) {
+        $query->the_post();
+        $post_id = get_the_ID();
+        $node_meta_data = get_post_meta($post_id,'tapestry_node_data', true);
+        if (isset($node_meta_data->mediaType) && $node_meta_data->mediaType == 'accordion') {
+            $node_meta_data->mediaType = 'multi-content';
+            $node_meta_data->presentationStyle = 'accordion';
+            update_post_meta( $post_id, 'tapestry_node_data',  $node_meta_data);
+        }
+    }
+}
+
+function accordion_row_update() {
+    $query_args = array(
+        'post_type' => 'tapestry_node',
+        'meta_query' => array(
+            array(
+               'key'     => 'tapestry_node_data',
+               'value'   => '"presentationStyle";s:13:"accordion-row"',
+               'compare' => 'LIKE',
+            ),
+        ),
+    );
+    $query = new WP_Query( $query_args );
+    while ( $query->have_posts() ) {
+        $query->the_post();
+        $post_id = get_the_ID();
+        $node_meta_data = get_post_meta($post_id,'tapestry_node_data', true);
+        if (isset($node_meta_data->presentationStyle) && $node_meta_data->mediapresentationStyleType == 'accordion-row') {
+            unset($node_meta_data->presentationStyle);
+            $node_meta_data->isMultiContentChild = true;
+            update_post_meta( $post_id, 'tapestry_node_data',  $node_meta_data);
+        }
+    }
+}
+
+function tapestry_plugin_update() {
+    global $TAPESTRY_VERSION_NUMBER;
+    $installed_version = get_site_option('tapestry_version');
+    if ( version_compare($installed_version, $TAPESTRY_VERSION_NUMBER, '<') ) {

```

Comment:
Nice usage of `version_compare`! 

So what I was saying here is that we should check the `$installed_version` against the specific version where this feature was introduced so that the update doesn't run every time there's a new version, but only once. For example, with your current code, if the `$installed_version` is 2.41.0 and the user updates to 2.44.0, it will update. But then again when the next update comes out (e.g. 2.45.0) `$installed_version` is 2.44.0 and so it will run this update again, and again on any future update. Not sure if this is what you intended, because then you need to make sure in your updates that you don't accidentally make assumptions about the data, etc. If you change to the following, then the update will only run once:
```suggestion
    if ( version_compare($installed_version, '2.44.0-beta', '<') ) {
```
Let me know if any of that doesn't make sense or if I missed something!

Reply:
Good call. Definitely what I intended but I hadn't considered that $TAPESTRY_VERSION_NUMBER would continue to update.